src/main/java/nextstep/ladder/domain/Ladder.java

```diff
@@ -1,19 +1,38 @@
 package nextstep.ladder.domain;
 
+import nextstep.ladder.strategy.DirectionStrategy;
+
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 public class Ladder {
     private final Lines lines;
 
+    public Ladder(int countOfPlayer, int ladderHeight, DirectionStrategy directionStrategy) {
+        this(new CountOfPlayer(countOfPlayer), new LadderHeight(ladderHeight), directionStrategy);
+    }
+
+    public Ladder(CountOfPlayer countOfPlayer, LadderHeight ladderHeight, DirectionStrategy directionStrategy) {
+        this(createWith(countOfPlayer, ladderHeight, directionStrategy));
+    }
+
+    private static Lines createWith(CountOfPlayer countOfPlayer, LadderHeight ladderHeight, DirectionStrategy directionStrategy) {
+        List<Line> lines = IntStream.range(0, ladderHeight.value())
+                .mapToObj((i) -> new Line(countOfPlayer, directionStrategy))
+                .collect(Collectors.toList());
+
+        return new Lines(lines);
+    }
+
     public Ladder(Lines lines) {
         this.lines = lines;
     }
```

Comment:
```suggestion
    private Ladder(Lines lines) {
        this.lines = lines;
    }
    
    public static Ladder of(Lines lines) {
        return new Ladder(lines);
    }

    public static Ladder of(CountOfPlayer countOfPlayer, LadderHeight ladderHeight, DirectionStrategy directionStrategy) {
        return of(Stream.generate(new Line(countOfPlayer, directionStrategy))
                .limit(ladderHeight.value())
                .collect(collectingAndThen(toList(), Lines::new)));
    }
    
    public static Ladder of(int countOfPlayer, int ladderHeight, DirectionStrategy directionStrategy) {
        return of(new CountOfPlayer(countOfPlayer), new LadderHeight(ladderHeight), directionStrategy);
    }
```

위와 같이 정적 팩토리 메소드 패턴을 이용하여 표현해주면 어떨까요!?
조금 개인차가 있긴 하지만 저는 생성자 혹은 정적팩토리 메소드 패턴을 사용할 때 위와 같은 순서로 작성합니다.
생성자 -> 기본 생성자를 사용하는 of -> 위에서 선언한 of를 사용하는 of  (반복)
그래야 코드의 흐름을 조금 더 쉽게 파악할 수 있는 것 같아요!

근데 앞서 말씀드리는 것 처럼 개인차가 존재하기 때문에 웬만하면 컨벤션을 정해놓고 만드는게 제일 좋긴 합니다 ㅎㅎ

Reply:
저는 부 생성자 ->  또 다른 부 생성자 -> 부 생성자를 주 생성자로 변환하는 정적 메소드 -> 주 생성자 순으로 정적 메소드를 최소화하기 위해서 해당 방식을 사용했었는데
말씀하신 방식대로 하게 되면 해당 객체의 주 생성자가 무엇인지 먼저 알게 되어 좋을 것 같네요! 감사합니다!